frontend/src/pages/enabledApplications/EnabledApplications.tsx

```diff
@@ -69,13 +50,32 @@ EnabledApplicationsInner.displayName = 'EnabledApplicationsInner';
 
 const EnabledApplications: React.FC = () => {
   const { components, loaded, loadError } = useWatchComponents(true);
+  const enabledComponents = React.useRef<ODHApp[]>([]);
```

Comment:
I tried to use `useRef` in this way but it didn't work. Every time you go back to the `Enabled` page, it will call `useWatchComponents` hooks, which will set the components to `[]`, then it sets the components again and fires the event. So `useRef` can only work if we keep staying at the `Enabled` page.

Reply:
RIght... Ok, we can put this back as an outer const

Reply:
Fixed

Reply:
@jeff-phillips-18 Sorry I cannot see the change of `enabledComponents`, could you update it again?

Reply:
sorry, file was missed. Fixed.